Practice 1

EmelyanovPavel opened the pull request.

tak333 commented on practice1-var3.cpp:
```diff
@@ -0,0 +1,309 @@
+//the following order : first, all characters other than digits, and then all digits,
+//maintaining the original order in each character group.
+
+void task2()
```
task1 и task2 в качестве исключений допускались в д/з.
Тут нужно показать именем, что ваша функция делает, по правилам хорошего кода.

tak333 reviewed (changes requested):
Не все замечания учтены.

Commit:
Выполнены все пункты задания на практику.

Commit:
Исправлен код

tak333 reviewed (changes requested):
Поясните ваш способ сохранения данных в список, извлечения из списка.
Проверьте подключаемые библиотеки.

EmelyanovPavel committed:
Update classQueue.h

Код без наследования абстрактного класса и переопределение его абстрактных функций без override

EmelyanovPavel committed:
Update practice-var3.cpp

Первая функция войд: изменен код добавления элементов в контейнер. Убраны объявленные переменные со значениями элементов контейнера.

tak333 reviewed (changes requested):
Не исправлены замечания, нет ответов на вопросы.

EmelyanovPavel commented:
я ответил на ваши вопросы

EmelyanovPavel committed:
Update classDoublyLinkedList.h

Реализован интерфейс и добавлен шаблон класса в скобки функции void add().

EmelyanovPavel committed:
Update classIContainer.h

убрано подключение файла с двусвязным списком

tak333 reviewed (changes requested):
Не исправлены замечания, неудовлетворительные ответы на вопросы.

tak333 commented:
Если вы согласны на "удовлетворительно", то работу приму.
Если нет, то учтите замечания.

tak333 commented:
Я не могу вытягивать из вас ответ на каждый вопрос, подготовьтесь: исправьте замечания, пересмотрите задание и ваш код, потом отправляйте запрос.

EmelyanovPavel committed:
Изменения в разделе public класса списка

В разделе public класса списка отредактирован обычный конструктор и добавлены конструктор с параметрами и конструкторы по правилу трёх.

EmelyanovPavel committed:
Изменения в коде

Защита файла от повторного включения

EmelyanovPavel committed:
Изменения в коде

Защита файла от повторного включения, класс-наследник Очередь от интерфейса, переопределение виртуальных функций абстрактного класса словом override.

EmelyanovPavel committed:
Изменения в коде

Убраны лишние библиотеки над функцией void output(), библиотеки для работы с файлом написаны над функцией void print().

EmelyanovPavel commented:
PS. В файле classDoublyLinkedList.h написана его защита от повторного включения

EmelyanovPavel committed:
Изменения в коде

Исправлена защита файла с классом очереди от повторного включения, где ифндеф и дефайн.

EmelyanovPavel committed:
Изменения в коде

Исправлена защита файла от повторного включения, где ифндеф, дефайн и эндиф.

tak333 reviewed (changes requested):
Приму на "удовлетворительно", либо исправляйте все замечания.

EmelyanovPavel committed:
Изменения в теле класса списка

Класс списка реализует двусвязный список с методами для добавления и удаления элементов с обоих концов, проверки на пустоту и очистки списка. Также реализованы копирующий конструктор и оператор присваивания для корректного управления памятью.

EmelyanovPavel committed:
Изменения в коде

Класс интерфейса сделан шаблонным

EmelyanovPavel committed:
Шаблонный класс очереди

Шаблонный класс очереди наследует интерфейс и реализует его методы, используя двусвязный список. Очередь поддерживает добавление элементов в конец и удаление из начала.

EmelyanovPavel committed:
Изменения в первой функции void.

В теле первой функции void изменен код для примера использования Очереди.

EmelyanovPavel commented:
Код работает правильно и без сбоев.

Выходные данные:
Default Constructor!
I'm queue
Adding elements...
Removing elements...
1 2 3
Destructor!
 e  s  d  g  g  j124637

EmelyanovPavel committed:
Изменения в теле класса списка

Изменены условия операторов if из методов добавления и удаления элементов и условия циклов while из копирующего конструктора и оператора присваивания, добавлены операторы сравнения в виде булевых функций.

EmelyanovPavel commented:
Код в остальных трёх файлах (classIContainer.h, classQueue.h и practice-var3.cpp) не изменён.

tak333 reviewed (approved):
Делайте отчёт.
